cdxev/amend/process_license.py

```diff
@@ -0,0 +1,215 @@
+
+    Parameters
+    ----------
+    :component: dict
+        A component
+    :license_name_id_map: list
+        A list with possible license names and
+        belonging to a license id
+    :path_to_license_folder: str (optional)
+        Path to a folder with txt files containing license texts
+
+    Returns
+    -------
+
+    """
+    licenses = component.get("licenses", [])
+    if not licenses:
+        return
+
+    component_id = ComponentIdentity.create(component, allow_unsafe=True)
+
+    for license in licenses:
+        if "license" not in license:
+            continue
+
+        current_license = license.get("license", {})
+        if "id" in current_license:
+            continue
+
+        replace_license_name_with_id(current_license, license_name_id_list)
+        add_text_from_folder_to_license_with_name(
+            current_license, path_to_license_folder, component_id
+        )
+
+    return
+
+
+def replace_license_name_with_id(license: dict, license_name_id_list: list) -> None:
+    """
+    Adds the id of a license to a license and removes the name, if the name
+    is in the list of licenses provided.
+
+    The operation is performed on the provided license.
+
+    Parameters
+    ----------
+    :license: dict
+        A license
+    :license_name_id_map: list
+        A list with possible license names and
+        belonging to a license id
+
+    Returns
+    -------
+
+    """
+    if "id" in license:
+        return
+
+    id_found = find_license_id(license.get("name", ""), license_name_id_list)
+    if id_found:
+        license["id"] = id_found
+        license.pop("name")
+    return
+
+
+def add_text_from_folder_to_license_with_name(
+    license: dict,
+    path_to_license_folder: str = "",
+    component_id: ComponentIdentity = ComponentIdentity.create({}, allow_unsafe=True),
+) -> None:
+    """
+    Adds the text describing a license,
+    if the provided folder contains a corresponding txt-file with the text of the license.
+    The txt-file has to follow the naming convention 'license name'.txt.
+
+    The operation is performed on the provided license.
+
+    Parameters
+    ----------
+    :license: dict
+        A license
+    :license_name_id_map: list
+        A list with possible license names and
+        belonging to a license id
+
+    Returns
+    -------
+
+    """
+    if path_to_license_folder and license.get("name", ""):
+        license_text = get_license_text_from_folder(
+            license.get("name", ""), path_to_license_folder
+        )
+        if license_text == "":
+            logger.warning(
+                LogMessage(
+                    "License text not found",
+                    (
+                        f"No text for the license ({license.get('name', '')}), "
+                        f"in component ({component_id}), was found. "
+                        "An empty string was added as text."
+                    ),
+                )
+            )
+        else:
```

Comment:
You will always print the warning now, even though you should only print it if license.get("text") != "". 
However, maybe it would be also a good addition to also have a `logger.info` that "the text of license xy" was included.